samcli/local/apigw/local_apigw_service.py

```diff
@@ -49,24 +57,27 @@ def __init__(
         :param str event_type: Type of the event. "Api" or "HttpApi"
         :param str payload_format_version: version of payload format
         :param bool is_default_route: determines if the default route or not
+        :param str stack_path: path of the stack the route is located
         """
         self.methods = self.normalize_method(methods)
         self.function_name = function_name
         self.path = path
         self.event_type = event_type
         self.payload_format_version = payload_format_version
         self.is_default_route = is_default_route
+        self.stack_path = stack_path
 
     def __eq__(self, other):
         return (
             isinstance(other, Route)
             and sorted(self.methods) == sorted(other.methods)
             and self.function_name == other.function_name
             and self.path == other.path
+            and self.stack_path == other.stack_path
         )
 
     def __hash__(self):
-        route_hash = hash(self.function_name) * hash(self.path)
+        route_hash = hash(f"{self.stack_path}-{self.function_name}-{self.path}")
```

Comment:
Not entirely familiar with this, but would this cause redeployments?

Reply:
A good question, I cannot find this hash is essentially used other than unit tests. 

First this Route's hash is used in `Api.__hash__` (samcli/lib/providers/provider.py:305), but then noone uses Api's hash.

Reply:
And since `Api` or `Route` is not buildable resources (no artifacts), I don't think they would cause redeployment. But the hash seems not used anymore, we can have a followup to try to remove them.